JourneysSquad/JourneysSquadTests/Model/BillStorage/DoubleLinkedListTests.swift

```diff
@@ -0,0 +1,138 @@
+import XCTest
+
+final class DoubleLinkedListTests: XCTestCase {
+    func testAppendingToTheEmptyList() {
+        var linkedList = DoubleLinkedList<Int>()
+
+        linkedList.append(value: 10)
+
+        XCTAssertEqual(linkedList.head?.value, 10)
+        XCTAssertEqual(linkedList.tail?.value, 10)
+    }
+
+    func testAppendingMultipleValues() {
+        var linkedList = DoubleLinkedList<Int>()
+
+        linkedList.append(value: 10)
+        linkedList.append(value: 5)
+        linkedList.append(value: 0)
+
+        XCTAssertEqual(linkedList.head?.value, 10)
+        XCTAssertEqual(linkedList.tail?.value, 0)
+    }
+
+    func testAppendingWithDifferentType() {
+        var linkedList = DoubleLinkedList<String>()
+
+        linkedList.append(value: "Apple")
+        linkedList.append(value: "Mango")
+        linkedList.append(value: "Lemon")
+
+        XCTAssertEqual(linkedList.head?.value, "Apple")
+        XCTAssertEqual(linkedList.tail?.value, "Lemon")
+    }
+
+    func testInsertionToTheEmptyList() {
+        var linkedList = DoubleLinkedList<Int>()
+
+        linkedList.insert(value: 5, before: nil)
+
+        XCTAssertEqual(linkedList.head?.value, 5)
+        XCTAssertEqual(linkedList.tail?.value, 5)
+    }
+
+    func testInsertionWithMultipleValuesWhenNodeBeforeInsertIsNil() {
+        var linkedList = DoubleLinkedList<Int>()
+
+        linkedList.append(value: 3)
+        linkedList.append(value: 7)
+        linkedList.append(value: 10)
+        linkedList.insert(value: 5, before: nil)
+
+        XCTAssertEqual(linkedList.head?.value, 5)
+    }
+
+    func testInsertionBeforeHead() {
+        var linkedList = DoubleLinkedList<Int>()
+
+        linkedList.append(value: 10)
+        guard let head = linkedList.head else {
+            XCTFail("Head node is nil")
```

Comment:
You should understand from message that this failure is not an ordinary case, that wasn't passed, but you have an internal problem in your code.
```suggestion
            XCTFail("Unexpected error.")
```